### PR TITLE
audio: beginnings of audio system

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

**Review by slimsag — APPROVED** [review]
Looks like a great starting point, thanks!

**Comment by alichraghi** [comment]
a lot of changes coming... :)

File: audio/build.zig [hunk 1]
```diff
@@ -0,0 +1,119 @@
+const std = @import("std");
+const Builder = std.build.Builder;
+
+const soundio_path = thisDir() ++ "/upstream/soundio";
+
+pub const pkg = std.build.Pkg{
+    .name = "soundio",
+    .source = .{ .path = thisDir() ++ "/soundio/main.zig" },
+};
```
Comment (slimsag) [inline audio/build.zig:9]: this should point to `audio/main.zig` right? We can do this in another PR, just making a note

File: audio/build.zig [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+pub const SoundIoOptions = struct {
+    jack: bool = false,
+    pulseaudio: bool = false,
+    alsa: bool = false,
+    coreaudio: bool = false,
+    wasapi: bool = false,
+};
```
Comment (slimsag) [inline audio/build.zig:17]: I'm not sure we should make these options public at all.

File: audio/build.zig [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    test_step.dependOn(&soundio_tests.step);
+}
+
+pub fn link(b: *Builder, step: *std.build.LibExeObjStep, options: SoundIoOptions) void {
+    const soundio_lib = buildSoundIo(b, step, options);
+    step.linkLibrary(soundio_lib);
+    step.addIncludePath(soundio_path);
+}
+
+pub fn buildSoundIo(b: *Builder, step: *std.build.LibExeObjStep, options: SoundIoOptions) *std.build.LibExeObjStep {
```
Comment (slimsag) [inline audio/build.zig:37]: I'd suggest we also do not make this public.

**Review by slimsag — APPROVED** [review]
Nice work!
